### ISSUE TITLE
[pickers] Implement out-of-the-box behavior to clear input

### ISSUE THREAD

**Issue by yanickrochon** [mui/mui-x#4450]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Summary 💡

I see this statement at multiple places in the documentation, but this feature seems to be completely missing in the code.

> **Note** that `showClearButton` has a higher priority.

### Examples 🌈

[Demo](https://codesandbox.io/s/basicdatepicker-material-demo-forked-4ndp2?file=/demo.js)

```jsx
// the clearable property does not change anything in the component
<DatePicker
        clearable
        showTodayButton
        label="Basic example"
        value={value}
        onChange={(newValue) => {
          setValue(newValue);
        }}
        renderInput={(params) => <TextField {...params} />}
/>
```

### Motivation 🔦

The ability to clear the field is trivial, but essential in certain cases.

**Comment by yanickrochon** [comment]
@flaviendelangle I eventually saw the clear button within the popup, but this is not what initially prompted me to create this issue. What prompted me was a failure to understand how MUI actually implements Material Design guidelines for picker components.

More specifically, the design describres two properties, first :

> **Icon signifier**
> Icons describe the type of input a text field requires. They are displayed to the left of the text field.
>
> Icons can also be touch targets for nested components. For example, a calendar icon may be tapped to reveal a date picker.

![image](https://user-images.githubusercontent.com/78461/163179934-5bec1f08-83b9-403f-8eb9-2f5dc114facc.png)

And, second :

> **Clear button**
> Clear buttons let users empty an entire input field.
>
> They are right justified in the field.

![image](https://user-images.githubusercontent.com/78461/163179861-fe1bcd3a-6d6a-4e0c-8fce-4b10095395f8.png)

However, MUI's design is this one :

![image](https://user-images.githubusercontent.com/78461/163179618-882cf78e-2148-4ab3-8de9-fb2be5abbf24.png)

I was expecting the input to be clearable without having to open the popup. I was expecting what the specification states.

**Comment by yanickrochon** [comment]
I understand, and I have noticed that very few UI frameworks actually follow the actual Material specifications in regards to user input elements. I have also noticed discrepancies between various component implementations as well (e.g. the [Select](https://mui.com/material-ui/react-select/) and [Autocomplete](https://mui.com/material-ui/react-autocomplete/) have very different APIs even if they mainly offer the same visual styles and UX)

In the case of of the Date/Time pickers, I was expecting the clear button to be visible without requiring extra steps to access the functionality. In my engineering classes (over a decade ago, perhaps, but I still remember the recommendations about heurisitcs and user interactions), this kind of design flaw was highlighted a few times.

But, most importantly, even MUI correctly implements the clear button elsewhere! The [Search](https://mui.com/material-ui/react-autocomplete/#search-input) input field does follow the specification :

![image](https://user-images.githubusercontent.com/78461/163189181-c28ce6b0-0a26-47dd-bf5c-8889764e88a7.png)

So, my point is, unless MUI actually follows a design standard, it will quickly become a mess and frustrate both users and devs. Points to consider.

**Comment by flaviendelangle** [comment]
For Select / Autocomplete, I think that Autocomplete is a lot more recent and that Select may evolve closer from the Autocomplete API in the future (don't take my word for it, I never worked on it).

____

For the reset button, I will have a look at the other component (thanks for the search example).
Switching the toggle icon to the left and add a reset icon should be easy.

____

For the following of the specification in general. The team behind https://github.com/mui-org/material-ui is building a new design system and the foundations to handle several design systems with the same unstyled bricks.
It is still not very clear how the our X components should play with those design systems, but we should aim at staying consistent. The pickers have been under-maintained because of the time required vs the time the team had, but we should be able to work a lot more on them now.

**Comment by zigang93** [comment]
it is possible to use `useRef` to picker and try to reset the value from picker? 
because I am trying to reset the `DateRangePicker` from other button with `onClick` method

**Comment by flaviendelangle** [comment]
@zigang93 could you give me a reproduction case ? 
And if this is not directly related to this issue, create a new one to keep the discussions separated :+1:

**Comment by yanickrochon** [comment]
@zigang93 take a look at issue #4479

**Comment by alexfauquette** [comment]
Just to bring some information about the clear button at the end of the search input. The cross you see is not implemented by MUI

If `<input/>` has that parameter `type="search"` chrome automatically adds a clear icon that is not an MUI icon. It can be targeted with `::-webkit-search-cancel-button` (most of the dime to remove it)

On [the same demo](https://mui.com/material-ui/react-autocomplete/#search-input) you can see that the `freeSolo` has the same behavior but here it is handled by MUI. The button has the same style as other <IconButton/>

From the behavior of `AutoComplet` we could do something like that:
https://codesandbox.io/s/basicdatepicker-material-demo-forked-i8jcm6?file=/demo.tsx

**Comment by mickbigblue** [comment]
Is there a way to get the clearable prop working in "DesktopDateTimePicker", too?
There is nothing in the API-Docs here:
https://mui.com/x/api/date-pickers/desktop-date-time-picker/

**Comment by flaviendelangle** [comment]
The demo from @alexfauquette above should probably work for `DesktopDateTimePicker` to
And for a built-in integration, if we do it for the `DatePicker` we should do it for others as well

**Comment by mickbigblue** [comment]
Ok, now I got it, there is no built-in integration via a simple `clearable` prop...

Do I also have to reference that 
```
    "@mui/x-date-pickers": "https://pkg.csb.dev/mui/mui-x/commit/fe427090/@mui/x-date-pickers",
```
in the package.json or can I just use the latest `@mui/x-date-pickers`?

**Comment by flaviendelangle** [comment]
You can use a recent version
This weird install line is to reference to a specific commit to demonstrate something not published yet, but it should not be used in production.

**Comment by mickbigblue** [comment]
All right. I'll look into it. Thanks!

**Comment by Cirelion** [comment]
Is there any update on a rough timeline for when this will be implemented? The Mui-lab datepicker has this functionality but I'd prefer to more over to mui-x as I'm getting deprecation warnings.

**Comment by Cirelion** [comment]
I'm talking about the clear icon inside the input on the right yes.
I can simply add a clearable tag to the @mui/lab/MobileDatePicker component. 
`<MobileDatePicker 
clearable 
...otherProps
/>`

And I'm well aware how to upgrade to mui-x but I don't want to before this feature is added :)

**Comment by LukasTy** [comment]
> any news about this feature?

Thank you for being interested in this feature. 🙏 
We are in the final stages of refining the PR. If all goes well, we hope to release it in the coming weeks. 🤞

**Comment by michelengelen** [comment]
@Lucas-Henrique-Lopes-Costa It will be included in the next release (6.16.0) 🚀
